src/CQRS/LeanCode.CQRS.OutputCaching/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,49 @@
+using System.Reflection;
+using LeanCode.Components;
+using LeanCode.CQRS.Execution;
+using LeanCode.CQRS.OutputCaching.Registration;
+using Microsoft.AspNetCore.OutputCaching;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace LeanCode.CQRS.OutputCaching;
+
+public static class ServiceCollectionExtensions
+{
+    public static IServiceCollection AddCQRSOutputCache(
+        this IServiceCollection services,
+        TypesCatalog catalog,
+        bool registerOutputCache = true
+    )
+    {
+        return services.AddCQRSOutputCache(catalog.Assemblies, registerOutputCache);
+    }
+
+    public static IServiceCollection AddCQRSOutputCache(
+        this IServiceCollection services,
+        IEnumerable<Assembly> assemblies,
+        bool registerOutputCache = true
```

Comment:
Do we need this flag? Can't we _always_ call `AddOutputCache`? AFAIK all the `IServiceCollection` extensions provided by ASP.NET are re-entrant, i.e. they use `TryAdd...` underneath, so they won't register things twice.

Reply:
You're probably right